build.py

```diff
@@ -141,7 +141,9 @@ def Run(context, cmd):
             while True:
                 l = p.stdout.readline().decode(encoding)
                 if l != "":
-                    logfile.write(l)
+                    # Avoid "UnicodeEncodeError: 'ascii' codec can't encode 
+                    # character" errors by serializing utf8 byte strings.
+                    logfile.write(l.encode("utf8"))
```

Comment:
This change is to fix the build.py errors reported by a few people.  Completely orthogonal to the rest of the pull request, but I ran into the problem while developing, so I fixed it.